### PR TITLE
link/kprobe: Allow `.` in symbol name

### PR DESCRIPTION
**Pull request by chenhengqi**
Signed-off-by: Hengqi Chen <chenhengqi@outlook.com>

### CODE REVIEW

**Comment by lmb** [comment]
Seems fine, could you take a look @mmat11 ?

File: link/kprobe.go [hunk 1]
```diff
@@ -141,7 +147,7 @@ func kprobe(symbol string, prog *ebpf.Program, opts *KprobeOptions, ret bool) (*
 	if prog == nil {
 		return nil, fmt.Errorf("prog cannot be nil: %w", errInvalidInput)
 	}
-	if !rgxTraceEvent.MatchString(symbol) {
+	if !rgxKprobe.MatchString(symbol) {
 		return nil, fmt.Errorf("symbol '%s' must be alphanumeric or underscore: %w", symbol, errInvalidInput)
```
Comment (mmat11) [inline link/kprobe.go:151]: nit: should we change the error message to a more generic one?

**Review by mmat11 — APPROVED** [review]

File: link/kprobe.go (hunk 1 above)
Comment (chenhengqi) [reply link/kprobe.go]: OK. Just updated.

**Review by mmat11 — APPROVED** [review]

**Review by lmb — APPROVED** [review]

**Comment by lmb** [comment]
Thanks!

**Comment by ti-mo** [comment]
Thanks!

**Pull request merged**